public/docs/_examples/toh-6/ts/app/hero-detail.component.ts

```diff
@@ -29,11 +31,11 @@ export class HeroDetailComponent implements OnInit {
   // #docregion save
   save(): void {
     this.heroService.update(this.hero)
-      .then(this.goBack);
+      .then(() => this.goBack());
```

Comment:
Oops, your change is actually the right thing to do to ensure that `this` gets bound properly in `goBack`.